packages/network-controller/src/NetworkController.ts

```diff
@@ -459,6 +468,12 @@ export class NetworkController extends BaseControllerV2<
     return isEIP1559Compatible;
   }
 
+  resetConnection() {
+    const { type, rpcTarget, chainId, ticker, nickname } =
+      this.state.providerConfig;
+    this.setupProvider(type, rpcTarget, chainId, ticker, nickname);
```

Comment:
What's the reason why here we call `this.setupProvider` instead of `this.initializeProvider`? 
We don't want to call `getIsCustomNetwork` or `getEIP1559Compatibility` for some reason?

Reply:
This is a good point. We should use `initializeProvider` here. The `isCustomNetwork` stuff will go away, but we do still want to call `getEIP1559Compatibility`.